Week 04/Lecture 08/Assignment 2/src/main/java/jebi/hendardi/repository/EmployeeRepository.java

```diff
@@ -0,0 +1,65 @@
+package jebi.hendardi.repository;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.stereotype.Repository;
+
+import jebi.hendardi.model.Employee;
+
+@Repository
+public class EmployeeRepository {
+
+    private final JdbcTemplate jdbcTemplate;
+
+    public EmployeeRepository(JdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    private final RowMapper<Employee> EMPLOYEE_ROW_MAPPER = new RowMapper<>() {
+        @Override
+        public Employee mapRow(ResultSet rs, int rowNum) throws SQLException {
+            return new Employee(
+                    rs.getString("id"),
+                    rs.getString("name"),
+                    rs.getDate("dob").toLocalDate(),
+                    rs.getString("address"),
+                    rs.getString("department")
+            );
+        }
+    };
+
+    public List<Employee> findAll() {
```

Comment:
Pls handle exception for methods.

Reply:
Exception handling has been added to the methods in EmployeeRepository.java to ensure that errors are caught and logged appropriately. This prevents the application from crashing due to unhandled exceptions and provides meaningful feedback in case of errors.

**Changes Made:**

- Wrapped database operations in try-catch blocks.
- Logged exceptions and provided default failure responses.